src/unitxt/metrics.py

```diff
@@ -6250,7 +6250,7 @@ def compute(self, references: List[Any], prediction: Any, task_data: Dict) -> di
         return result
 
     def create_message(self, role: str, content: str) -> List[Dict[str, str]]:
-        return [{"role": role, "content": content}]
+        return [{"role": role, "content": str(content)}]
```

Comment:
From your changes, we are already checking that the content is a string (for call and tool fields), so I don't think a conversion to string is required.